### PR TITLE
Add `.inl` extension

### PR DESCRIPTION
**Pull request by PointKernel**
This PR adds `.inl` as a c++ file extension.

### CODE REVIEW

**Comment by asottile** [comment]
this should be two separate PRs

**Comment by PointKernel** [comment]
I haven't found any solid references yet but using `.inl` headers is a common practice to separate template/inline interface from implementation in c++ headers. Here are some links you may find useful:

1. [Stackoverflow discussion on "Significance of a .inl file in C++"](https://stackoverflow.com/questions/1208028/significance-of-a-inl-file-in-c/1208062)
2. [INL file definition from fileinfo.com](https://fileinfo.com/extension/inl)

File: identify/extensions.py [hunk 1]
```diff
@@ -80,6 +80,7 @@
     'idr': {'text', 'idris'},
     'inc': {'text', 'inc'},
     'ini': {'text', 'ini'},
+    'inl': {'text', 'c++'},
```
[the remark this answers is not in the document]
Comment (PointKernel) [reply identify/extensions.py]: Done

**Review by asottile — APPROVED** [review]

**Pull request merged**